lib/Doctrine/ORM/EntityManagerInterface.php

```diff
@@ -80,12 +80,30 @@ public function beginTransaction();
      * If an exception occurs during execution of the function or flushing or transaction commit,
      * the transaction is rolled back, the EntityManager closed and the exception re-thrown.
      *
+     * @deprecated 2.10 Use {@link wrapInTransaction} instead.
+     *
      * @param callable $func The function to execute transactionally.
      *
      * @return mixed The non-empty value returned from the closure or true instead.
      */
     public function transactional($func);
 
+    /**
+     * Executes a function in a transaction.
+     *
+     * The function gets passed this EntityManager instance as an (optional) parameter.
+     *
+     * {@link flush} is invoked prior to transaction commit.
+     *
+     * If an exception occurs during execution of the function or flushing or transaction commit,
+     * the transaction is rolled back, the EntityManager closed and the exception re-thrown.
+     *
+     * @param callable $func The function to execute transactionally.
+     *
+     * @return mixed The value returned from the closure.
+     */
+    public function wrapInTransaction(callable $func);
```

Comment:
@SenseException maybe let's just do this:
```suggestion
    // public function wrapInTransaction(callable $func);
```

callers will have to check that this method exists before calling it though.

Reply:
This will do it for 2.10.

Reply:
or I can target 3.0?

Reply:
@simPod please don't, that would mean we cannot remove `transactional` in 3.0

Reply:
I'm waiting for instructions 🙃

Reply:
Ok, here are the steps:

1. apply my suggestion
2. squash it with the previous commit
3. improve the upgrade notes by explaining how people are supposed to check that `wrapInTransaction` exists before calling it, if they have an instance of `EntityManagerInterface`

Reply:
I'm really curious. What is the reason to comment this out? Why can't it be the interface?